### PR TITLE
chore: consolidate AI agent config under .github/

### LINKED ISSUE #1085: chore: remove low-value AI editor rules and chatmodes
## Summary

Several AI editor rule files and chatmodes provide no value or actively cause harm. They should be removed.

### Harmful (causes repo corruption)

Already tracked in #1084:
- `.cursor/rules/codacy.mdc`
- `src/tools/PerfDiff/.cursor/rules/codacy.mdc`

### Harmful (reckless autonomous behavior)

- **`.github/chatmodes/Beast Mode.chatmode.md`**: GPT-4.1 chatmode that instructs the AI to "keep going until solved" without user approval, recursively fetch web pages, and never ask the user for input. Contradicts the project's deliberate, no-trial-and-error, interactive approach. High risk of autonomous damage similar to the Codacy rule cascade.

### Zero value (burns context for no benefit)

- **`.cursor/rules/verify-information-rule.mdc`**: One-line rule with `alwaysApply: true` saying "verify information before presenting it." Too vague to be actionable. Wastes context tokens on every Cursor interaction.
- **`.cursor/rules/conventional-commit-messages.mdc`**: 50-line Conventional Commits spec with `alwaysApply: true`. Already enforced by the pre-commit hook and CI. Duplicates built-in AI knowledge. Wastes ~700 tokens of context on every Cursor interaction.

### Low value (optional, consider removing)

- **`.cursor/rules/csharp.mdc`**: One-line redirect to `copilot-instructions.md`. Unnecessary indirection.
- **`.windsurf/rules/csharp.md`**: Same one-line redirect for Windsurf. Same assessment.

## Files to Keep

- `.cursor/rules/create-explainer.md`: Useful PRD template, `alwaysApply: false`
- `.cursor/rules/process-task-list.md`: Useful process doc, not auto-applied
- `.githooks/pre-commit`: Well-scoped, valuable
- `.githooks/pre-push`: Valuable (separate bug in #1081 for Scan-TodoComments.ps1 parse failure)

## Rationale

AI editor rules with `alwaysApply: true` consume context tokens on every interaction. Rules that instruct autonomous "fix and continue" behavior risk cascading file corruption (proven by ~110 files damaged from the Codacy rule). Rules that duplicate CI enforcement or built-in AI knowledge waste context without adding value.

### PR DESCRIPTION
**Pull request by rjmurillo**
## Summary

Standardizes on **GitHub Copilot + Copilot CLI** and consolidates all AI/agent instructions and configuration under `.github/`. Resolves #1085.

### `.github/` is now the single home for AI config
- `.github/copilot-instructions.md` — main instructions (references updated)
- `.github/instructions/**` — file-type instructions (unchanged)
- `.github/prompts/*.prompt.md` — **new**: migrated workflow prompts (`create-explainer`, `generate-tasks`, `process-task-list`) with Copilot `mode: agent` frontmatter
- `.github/mcp.json` — **new**: Copilot CLI MCP config (serena, sequential-thinking, codacy, supermemory; the GitHub MCP server is built into Copilot CLI, so it is omitted)

### Removed (non-Copilot / harmful — #1085)
- `.cursor/` (Cursor rules incl. the corruption-causing `codacy.mdc`)
- `.windsurf/` (Windsurf rules)
- `.github/chatmodes/Beast Mode.chatmode.md` (reckless autonomous behavior)
- `src/tools/PerfDiff/.cursor/` and the now-empty `src/tools/PerfDiff/.gitignore`
- root `.mcp.json` (Claude-only; replaced by `.github/mcp.json`)

### Kept intentionally
- `.vscode/mcp.json` + `.vscode/settings.json` (VS Code editor MCP)
- `.serena/memories/` and root `AGENTS.md` (thin pointer)

## Why `.github/mcp.json` (verified, not assumed)

Inspecting the **Copilot CLI 1.0.57** bundle confirms the loader:

```js
yua = ".mcp.json"; bua = ".github/mcp.json";
for (let n of [yua, bua]) { /* read first one found */ }
// .vscode handling is warn-only:
"Copilot CLI's incomplete support for .vscode/mcp.json has been removed.
 See https://gh.io/copilotcli-mcpmigrate to migrate to .mcp.json."
```

- Copilot CLI loads project MCP config from `.mcp.json` → `.github/mcp.json` (first match wins).
- `.vscode/mcp.json` is **no longer read** by the CLI (it only serves the VS Code editor now).
- Root `.mcp.json` is removed so it cannot shadow `.github/mcp.json`.

> Note: keeping `.vscode/mcp.json` means the CLI shows a one-time "migrate to .mcp.json" notice. This is an accepted trade-off to retain VS Code editor MCP support.

## Validation

- Pre-commit hooks passed: markdownlint, yamllint, actionlint, json-validate, large-file-detection.
- Pre-push hook passed: full `dotnet build` + test suite (deterministic CI config).
- `.github/mcp.json` validated as JSON; migrated prompts validated with `markdownlint-cli2`.
- No remaining references to removed paths (only intentional historical lessons in Serena memories).
- No `.cs`/analyzer behavior changes; Moq version compatibility unaffected (docs/config only).

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

## Summary by CodeRabbit

* **Chores**
  * Consolidated AI editor guidance and MCP server setup under the GitHub configuration area, updating Copilot integration guidance.
  * Removed legacy editor-specific rule files and deprecated the previous Serena MCP configuration entry.
  * Adjusted CI linter path exclusions to match the reorganized structure.

* **Documentation**
  * Updated AI editor rules documentation to reflect the new consolidated locations.
  * Added agent metadata to task/prompt templates and refreshed task-processing prompt wording and references.

### CODE REVIEW

File: .github/mcp.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      "command": "uvx",
+      "args": [
+        "--from",
+        "git+https://github.com/oraios/serena@v0.1.4",
+        "serena",
+        "start-mcp-server",
+        "--context",
+        "ide-assistant",
+        "--project",
+        "${workspaceFolder:-.}"
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/mcp.json]: Fixed in 0b84286.

Changed `"${workspaceFolder:-.}"` to a literal `"."` for the serena `--project` argument.

Context for traceability: I verified against the Copilot CLI 1.0.57 bundle's variable expander. It only substitutes the exact token `${workspaceFolder}`; the `${name:-default}` form is handled by a separate rule that treats `workspaceFolder` as an environment variable and falls back to the default. Since that env var is unset, `${workspaceFolder:-.}` already resolved to `.`. The literal `"."` is equivalent, clearer, and correct because the CLI runs from the repository root. Thanks for the catch.

[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/mcp.json]: Already addressed on the current head: `.github/mcp.json` uses `"."` for Serena's `--project` arg (line 14). No `${workspaceFolder:-.}` placeholder remains. Resolving.

File: .github/prompts/generate-tasks.prompt.md [hunk 2]
```diff
@@ -65,7 +70,7 @@ Parent: #123
 
 ### Notes
 
-When editing files, follow the guidance at `.github/instructions/README.md` to determine appropriate instructions for specific files.
+When editing files, follow the guidance at [instructions/README.md](../instructions/README.md) to determine appropriate instructions for specific files.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/prompts/generate-tasks.prompt.md]: Fixed in 850e3be. The template link is now root-relative (`.github/instructions/README.md`) so it resolves when the Notes block is copied into a GitHub issue body. Resolving.

File: .github/prompts/process-task-list.prompt.md [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
+         git commit -m "feat: add payment validation logic" -m "- Validates card type and expiry" -m "- Adds unit tests for edge cases" -m "Related to #123 in Explainer" -m "Fixes sub-task #456"
+         ```
 
   3. Once all the subtasks are marked completed and changes have been committed, verify with the user the task is completed.
   4. Once the user has indicated the work is verified, push the branch and open a pull request.
-    - **Title**: Uses convention commit format
-    - **Body**: Be descriptive
-      - **Explain**: all changes made, why they were made, and all validation performed
-      - **Reference the Task** Use language to indicate the issue is resolved at the end of the description (e.g., `Fixes #456`, `Closes #123`, `Resolves #789` etc. )
+     - **Title**: Uses convention commit format
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/prompts/process-task-list.prompt.md]: Fixed in 850e3be: changed to the canonical "Conventional Commits". Resolving.

File: .github/mcp.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        "ide-assistant",
+        "--project",
+        "."
+      ],
+      "tools": ["*"]
+    },
+    "sequential-thinking": {
+      "type": "stdio",
+      "command": "npx",
+      "args": ["-y", "@modelcontextprotocol/server-sequential-thinking"],
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/mcp.json]: Valid. Serena is pinned (`@v0.1.4`) but the three `npx -y` MCP packages (`@modelcontextprotocol/server-sequential-thinking`, `@codacy/codacy-mcp`, `supergateway`) float. Deferring the pin to a follow-up so exact versions are chosen deliberately rather than guessed. Leaving this thread open to track it.

[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/mcp.json]: Fixed in e7e7e3e. Pinned exact versions (verified as current published releases on npm): `@modelcontextprotocol/server-sequential-thinking@2025.12.18`, `@codacy/codacy-mcp@0.6.22`, and `supergateway@3.4.3`. serena was already pinned via `git+https://github.com/oraios/serena@v0.1.4`.
